### PR TITLE
fix race condition from multiple reads of HTMLMediaElement.buffered

### PR DESCRIPTION
**Pull request by kelpycreek**
Reading directly from target.buffered creates a race condition where its value can change if the user is clicking backwards through the video a lot. This should resolve that.

### CODE REVIEW

File: src/js/mep-feature-progress.js [hunk 1]
```diff
@@ -247,8 +247,10 @@
 
 			// newest HTML5 spec has buffered array (FF4, Webkit)
 			if (target && target.buffered && target.buffered.length > 0 && target.buffered.end && target.duration) {
+				//reading target.buffered can lead to race conditions
```
Comment (mziwisky) [inline src/js/mep-feature-progress.js]: i think this comment is hard to make sense of out of the context of this diff.  it looks like you're saying something can lead to race conditions, and then you're doing that thing anyway.  so for a comment, i think something like "target.buffered can change in the middle of this execution thread, so we must store a reference to it to make sure we're dealing with the exact same object below" is more descriptive

but really, i'd just omit the comment altogether.  race condition or not, it's a nicer style anyway to peel off that property and make line 253 more succinct, so it doesn't really need justification with a comment.

File: src/js/mep-feature-progress.js [hunk 2]
```diff
@@ -247,8 +247,10 @@
 
 			// newest HTML5 spec has buffered array (FF4, Webkit)
 			if (target && target.buffered && target.buffered.length > 0 && target.buffered.end && target.duration) {
```
Comment (mziwisky) [inline src/js/mep-feature-progress.js]: actually, we should peel buffer off of target even before this conditional

**Review by mziwisky — CHANGES_REQUESTED** [review]
don't forget to build the package (`grunt html5only`) and add the built assets to the commit

File: src/js/mep-feature-progress.js [hunk 3]
```diff
@@ -246,9 +246,10 @@
 				percent = null;
 
 			// newest HTML5 spec has buffered array (FF4, Webkit)
-			if (target && target.buffered && target.buffered.length > 0 && target.buffered.end && target.duration) {
-				// account for a real array with multiple values - always read the end of the last buffer
-				percent = target.buffered.end(target.buffered.length - 1) / target.duration;
+			const buffer = target && target.buffered; 
+			if (buffer && target.buffered.length > 0 && target.buffered.end && target.duration) {
```
Comment (mziwisky) [inline src/js/mep-feature-progress.js]: still referencing `target.buffered` a couple of times here.  should just deal with the `buffer` reference instead

**Review by mziwisky — CHANGES_REQUESTED** [review]

**Review by mziwisky — APPROVED** [review]
good stuff

**Comment by bensolo-inst** [comment]
Looks good.  No issues observed.  Tested in:  Chrome, Firefox, Safari, and Edge   (Though SauceLabs performance in Edge was very poor today / difficult to test well in local env.)

**Pull request merged**